### ISSUE TITLE
No Javadoc under download.eclipse.org

### ISSUE THREAD

**Issue by kolovos** [eclipse/epsilon-website#30]
The following URLs that used to contain the Javadoc of Epsilon's latest stable/interim version, and which are linked from the left-hand side tree in the website, no longer seem to be working:

- https://download.eclipse.org/epsilon/stable-javadoc/
- https://download.eclipse.org/epsilon/interim-javadoc/

Ideally, we should provide Javadoc for all (recent) versions of Epsilon, not only the latest stable and interim.

**Comment by agarciadom** [comment]
I have just pushed a potential fix. It looks like our `mvn -B initialize javadoc:aggregate` command was silently failing, passing the build even though the javadoc command failed. It was crashing due to somehow not having access to JUnit while parsing our tests sources, with errors like these:

```
/home/agd516/Documents/org.eclipse.epsilon/tests/org.eclipse.epsilon.etl.engine.test.acceptance/src/org/eclipse/epsilon/etl/engine/test/acceptance/tree2graph/Tree2GraphTest.java:31: error: cannot find symbol
	@Parameters(name = "{0}")
	            ^
  symbol:   method name()
  location: @interface Parameters
```

I've changed it to focus on the `plugins` folder, which does not exhibit these issues. I assume that there is little value in producing javadocs for our internal tests.

**Comment by agarciadom** [comment]
https://download.eclipse.org/epsilon/interim-javadoc/ is working as of 09b78babae5580856f2c729b86f7a80bf42e3ceb. 

I've made more tweaks to unify the m-javadoc-p versions and configuration in 644e96ad7dd436201180c8a91051aaa5984cae97, so the CI build will fail if the javadocs cannot be built (hopefully reducing the likelihood of problems like this in the future).

The difficult part is going to be fixing the javadocs for older versions. I suggest we start by creating a branch off the `2.5.0` tag (say, `2.5.0-javadoc`), and changing the `Jenkinsfile` in it so we focus on the `plugins` again and we do not upload anything except the fixed Javadocs.

If this works, we could do something similar for the other 2.x releases, and I'd suggest changing the `interim-javadoc` and `stable-javadoc` for just `X.Y.Z-javadoc`. This would reduce the work involved in having versioned javadocs over the years.

Does this sound reasonable, @kolovos?

**Comment by kolovos** [comment]
Yes, that sounds like a good plan to me!

**Comment by agarciadom** [comment]
The interim and stable Javadocs are now available from their usual URLs, and we've also got version-based URLs now:

https://download.eclipse.org/epsilon/2.5.0-javadoc/
https://download.eclipse.org/epsilon/2.6.0-javadoc/

**Comment by agarciadom** [comment]
We've got a problem for 2.4.0 and earlier: their target platforms depend on the GMF Tooling update site, which no longer exists. See [this failed build](https://ci.eclipse.org/epsilon/job/interim-kubernetes/job/2.4.0-javadoc/1/execution/node/42/log/):

```
[ERROR] Failed to resolve target definition file:/home/jenkins/agent/workspace/interim-kubernetes_2.4.0-javadoc/releng/org.eclipse.epsilon.target/org.eclipse.epsilon.target.target: Failed to load p2 metadata repository from location http://download.eclipse.org/modeling/gmp/gmf-tooling/updates/releases: No repository found at http://download.eclipse.org/modeling/gmp/gmf-tooling/updates/releases. -> [Help 1]
```

We could perhaps create javadocs out of the plain Maven build, if you like?

**Comment by agarciadom** [comment]
This should be fixed from 43958fb9447e2ff6f150f588d32120c5baf88661. We've got Javadocs for every version from 2.1.0 onwards. I had to exclude Eugenia, however (no GMF Tooling site), and for 2.1.0 I had to exclude picto.plantuml as `javadoc` was failing to parse some of the classes.
